lattices/LatticeMath/LatticeFFT.h

```diff
@@ -64,48 +66,66 @@ template <class T> class Lattice;
 
 class LatticeFFT
 {
+
 public: 
   // 2-D in-place complex->complex FFT. Transforms over the first two
   // dimensions and iterates over all the others. The Lattice must have two or
   // more dimensions otherwise an AipsError is thrown.
-  static void cfft2d(Lattice<DComplex> & cLattice, const Bool toFrequency=True);
-  static void cfft2d(Lattice<Complex> & cLattice, const Bool toFrequency=True);
+    template <class ComplexType> static void cfft2d(
+        Lattice<ComplexType> & cLattice, const Bool toFrequency=True
+    );
   
   // N-D in-place complex->complex FFT. Only transform over selected
   // dimensions. Iterate over the others. whichAxes must be the same length as
   // the number of dimensions in the Lattice otherwise an AipsError is thrown.
-  static void cfft(Lattice<DComplex> & cLattice,
+    template <class ComplexType> static void cfft(Lattice<DComplex> & cLattice,
   		  const Vector<Bool> & whichAxes, const Bool toFrequency=True);
-  static void cfft(Lattice<Complex> & cLattice,
+    template <class ComplexType> static void cfft(Lattice<ComplexType> & cLattice,
   		  const Vector<Bool> & whichAxes, const Bool toFrequency=True);
 
   // Non-folded version
-  static void cfft0(Lattice<Complex> & cLattice,
+    template <class ComplexType> static void cfft0(Lattice<ComplexType> & cLattice,
   		  const Vector<Bool> & whichAxes, const Bool toFrequency=True);
 
   // N-D in-place complex->complex FFT. Transform over all axes.
-  static void cfft(Lattice<DComplex> & cLattice, const Bool toFrequency=True);
-  static void cfft(Lattice<Complex> & cLattice, const Bool toFrequency=True);
+    template <class ComplexType> static void cfft(
+        Lattice<DComplex> & cLattice, const Bool toFrequency=True
```

Comment:
Does the cLattice argument have the correct type here? Or do you even need both this method and the following one?

Reply:
Good find. Looks like I missed that one and also a couple more. I'll fix that up. Thanks.